dulwich/porcelain.py

```diff
@@ -1753,6 +1755,136 @@ def update_head(repo, target, detached=False, new_branch=None):
             r.refs.set_symbolic_ref(b"HEAD", to_set)
 
 
+def unstage(repo: Repo, paths: list[bytes] = []):
+    '''
+    unstage specific file in the index
+    Args:
+        repo: dulwich Repo object
+        paths: a list of file to unstage
+    '''
+    for path in paths:
+        index = repo.open_index()
+        tree_id = repo[b'HEAD']._tree
+        try:
+            tree_entry = repo[tree_id].lookup_path(lambda x: repo[x], path)
+        except KeyError:
+            # if tree_entry didnt exist, this file was being added, so remove index entry
+            try:
+                del index[path]
+                index.write()
+            except KeyError:
+                print('file not in index.', path.decode())
+            return
+
+        try:
+            index_entry = list(index[path])
+        except KeyError:
+            # if index_entry doesnt exist, this file was being removed. readd it
+            # update index entry stats to reflect commit
+            index_entry = [(0, 0), (0,0), 15, 0, 0, 1000, 1000, 0, tree_entry[1], 0, 0]
+        index_entry[0] = (repo[b'HEAD'].commit_time, 0)  #ctime
```

Comment:
index_entry is a namedtuple nowadays, so you should be able to use index_entry.ctime here - which is slightly more readable.